.github/workflows/main.yml

```diff
@@ -5,25 +5,35 @@ on:
 
 jobs:
   build:
-    name: build
-    runs-on: ubuntu-latest
-
+    strategy:
+      matrix:
+        include:
+          - name: Ubuntu latest
+            os: ubuntu-latest
+          - name: macOS latest
+            os: macos-latest
+    name: ${{ matrix.name }}
+    runs-on: ${{ matrix.os }}
     steps:
-      - uses: actions/checkout@v4
-
-      - name: Install dependencies
-        run: |
-          sudo apt-get update
-          sudo apt-get install --yes \
-            libswresample-dev libavformat-dev libavutil-dev libavcodec-dev libfftw3-dev
-
-      - name: Build libkeyfinder
-        run: |
-          git clone https://github.com/mixxxdj/libkeyfinder keyfinder
-          cd keyfinder
-          cmake -S . -B build
-          cmake --build build
-          sudo cmake --install build
-
-      - name: Build keyfinder-cli
-        run: make
+    - name: Check out Git repository
+      uses: actions/checkout@v4
+    - name: "[Ubuntu] Install dependencies"
+      if: startsWith(matrix.os, 'ubuntu')
+      run: |
+        sudo apt-get update
+        sudo apt-get install --yes \
+          libswresample-dev libavformat-dev libavutil-dev libavcodec-dev libfftw3-dev
+    - name: "[macOS] Install dependencies"
+      if: startsWith(matrix.os, 'macos')
+      run: |
+        brew install ffmpeg libkeyfinder make
+    - name: "[Ubuntu] Build libkeyfinder"
+      if: startsWith(matrix.os, 'ubuntu')
+      run: |
+        git clone https://github.com/mixxxdj/libkeyfinder keyfinder
+        cd keyfinder
+        cmake -S . -B build
+        cmake --build build
+        sudo cmake --install build
```

Comment:
Would actually maybe make sense to split this into a totally different workflow since there's so many conditions?

Reply:
Yes, the conditions are not very nice.  I've now split this into two different jobs (in the same workflow file), which makes it more readable.